ls1intum/Artemis
# `Discussion:` Fix minor style and usability issues

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


I had some minor To-Dos on my list that I want to address all at once to avoid testing overhead.

### Description

Changes:
**Client**
- Green of check-mark to mark answer posts as resolving and posts as resolved is lighter, in oder to better distinguish to to the gray check-mark (not yet marked as resolved)
- Some style files were missing the `component` part in the file name, I added it now
- Some minor padding inconsistencies
- The displayed names for sort options are not a bit shorter to increase the width of the dropdown menu -> linebreak occurs a bit later (w.r.t to decreasing screen size)
- The logic for triggering the search for similar posts was changed:
1. We subscribe only to changes on the title form group field to trigger the search as this is the only field required at the moment
2. We do not send a request to the server, if the title input field is empty but rather set the array of similarPosts to an empty array -> the accordion of similar posts is not shown at all 
3. When submitting a post and click on the button to create a new one afterwards the similar posts accordion showed the similar posts that were shown on the previous post creation, this is now fixed -> no accordion

**Server**
- Endpoints for updating posts / answerPosts did not contain the id of the posting, this does not conform to our code guidelines as @PatZintel pointed out to me. I changed those two endpoints accordingly.

### Steps for Testing


As any User:
1. Log in to Artemis
2. Navigate to course discussion overview
3. Check if the sort criterion dropdown options are correct in English and German
4. Find a post that the currently logged in user posts
5. Toggle (or create) the answer -> click the checkmark to mark answer as resolving, check if the green color is distinguishable from the grey (not marked as resolved)
6. Make sure that what is shown in the video below **Similarity Check on Post Creation, After** looks the same for you.

As Student:
1. Edit a post and answer post that you created -> should work
2. Check that you cannot edit or delete posts / answer post from other students
3. Check that you cannot create an announcement (select announcement as course-wide topic)
4. Check that you do not see the pin and archive options next to the reaction selection in the post footer

As Instructor:
1. Edit a post and answer post that you did not create -> should work
2. Check that you can edit or delete posts / answer post from other students
3. Check that you can create an announcement (select announcement as course-wide topic)
4. Check that you do see the pin and archive options next to the reaction selection in the post footer

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


**Course discussion Interaction Bar:**
Before

<img width="1131" alt="Bildschirmfoto 2021-10-27 um 14 42 25" src="https://user-images.githubusercontent.com/41366522/139067664-e4d95915-9526-4d90-83e7-6fa35d747028.png">
After
<img width="1142" alt="Bildschirmfoto 2021-10-27 um 14 42 10" src="https://user-images.githubusercontent.com/41366522/139067683-3210f7d7-692a-436d-9a1b-1cd69f5b29ee.png">

**Post / Answer Post "Resolved Green":**
**Before**
<img width="1101" alt="Bildschirmfoto 2021-10-27 um 14 45 02" src="https://user-images.githubusercontent.com/41366522/139068056-0c79c959-a4a8-42fa-ab5f-75fb6d276405.png">

**After**
<img width="1097" alt="Bildschirmfoto 2021-10-27 um 14 44 24" src="https://user-images.githubusercontent.com/41366522/139068089-2a3ba4e5-a67a-4eda-ad48-dc293657a5e2.png">

**Similarity Check on Post Creation**
**Before**

https://user-images.githubusercontent.com/41366522/139070851-473fa057-f37a-4da0-b300-d6e0a1f7ac40.mov

**After**

https://user-images.githubusercontent.com/41366522/139070741-33624f27-4974-453b-a25c-e13d43622116.mov
